brew/tailscale.rb

```diff
@@ -0,0 +1,87 @@
+# typed: false
+# frozen_string_literal: true
+
+# TODO(mkramlich): generate from template, esp url/tag/rev, version, sha256
+
+# Homebrew formula for Tailscale
+class Tailscale < Formula
+  desc "Easiest, secure, crossplatform way to use WireGuard w/oauth2, 2FA/SSO"
+  homepage "https://www.tailscale.com"
+  # url "https://github.com/tailscale/tailscale/archive/v1.2.0.tar.gz"
+  # sha256 "7af324d3d1cdae6b71e3ecbbbc3240359b053530e388d539b4a45405c8ceeb9e"
+  # url "https://github.com/tailscale/tailscale/archive/v1.4.4.tar.gz"
+  # sha256 "5312c6d075a32049912e0932a89269869def9ac8ea9d0fdccc6b41db60fc2d4c"
+  # url "http://localhost:8901/tailscale/tailscale/archive/v1.5.0.tar.gz"
+  # sha256 "1b3843a4196ab9636adeefd9ef40b48494ebb3d19319c94d559ffda0a483b3c2"
+  # sha256 "a62caf2eb5c84d2d1775cd8a17eeb0e8ad3b8dbc9453399862d908227e0af721"
+  url "https://github.com/tailscale/tailscale.git",
+      revision: "7ad3af21417f0caa83e4e4542cc9571fec22e521" # , tag:      "v1.5.0"
+  version "1.5.0" # this is not truly 1.5.0 but brew required a version
+  license "MIT" # TODO(mkramlich): right license?
+  head "https://github.com/tailscale/tailscale.git",
+       branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    tailscale_path = buildpath/"src/github.com/tailscale/tailscale"
+    tailscale_path.install buildpath.children
+    # system "echo buldpath: #{buildpath}"
+    # system "echo tailscale_path: #{tailscale_path}"
+    cd tailscale_path do
+      system "go", "build", "-o", ".", "tailscale.com/cmd/tailscale"
+      system "go", "build", "-o", ".", "tailscale.com/cmd/tailscaled"
+      # system "ls -la"
+      bin.install "tailscale"
+      bin.install "tailscaled"
+    end
+  end
+
+  def post_install
+    (var/"run/tailscale").mkpath
+    (var/"lib/tailscale").mkpath
+  end
+
+  plist_options manual: "sudo tailscaled --socket=#{HOMEBREW_PREFIX}/run/tailscale/tailscaled.sock --state=#{HOMEBREW_PREFIX}/lib/tailscale/tailscaled.state --tun=utun5"
+
+  def plist
```

Comment:
I don't think this is needed, tailscaled has the ability to install its own launchd agent as of https://github.com/tailscale/tailscale/commit/29b028b9c42321621044578daac71b38fd3b9d92. You should probably put your launchd enhancements into cmd/tailscaled/install_darwin.go and add a caveat explaining how those two commands work.

Reply:
ok thanks. due to quirks in the space-time continuum my feature actually preceded that one, haha. but yes I was aware of it and chatted with Brad about it some. I have a strategy in mind to address this, but I think its best for me to address it in docs and working code. I'll address better in upcoming revs.